src/core/Akka/Actor/Props.cs

```diff
@@ -244,8 +244,11 @@ public bool Equals(Props other)
         {
             if (ReferenceEquals(null, other)) return false;
             if (ReferenceEquals(this, other)) return true;
-            return CompareDeploy(other) && CompareSupervisorStrategy(other) && CompareArguments(other) &&
-                   CompareInputType(other);
+            return Type == other.Type
+                && _producer.GetType() == other._producer.GetType()
```

Comment:
I added here a simple producer type compare, there was  no comparison on the producer before

Reply:
One other idea would be to let test if the producer implements `IEqualityComparer<Akka.Props>` 
or maybe add a field for it into the producer interface.